### ISSUE TITLE
Is it feasible to use a zed binocular camera on jetson nano?

### ISSUE THREAD

**Issue by Shame-fight** [AlexeyAB/darknet#5047]
Is it possible to run yolov3-tiny with a zed stereo camera on nano, outputting the distance of the target
Like this LD_LIBRARY_PATH =. /: $ LD_LIBRARY_PATH ./uselib data / coco.names cfg / yolov3-tiny.cfg yolov3-tiny.weights zed_camera

**Comment by AlexeyAB** [comment]
Yes, it should work: https://www.stereolabs.com/blog/announcing-zed-sdk-for-jetson-nano/

**Comment by Shame-fight** [comment]
> Yes, it should work: https://www.stereolabs.com/blog/announcing-zed-sdk-for-jetson-nano/

Hi,I have compiled and passed darknet on jetson nan. Although there are many warnings, it is successful. The uselib file is generated. My makefile is as follows:

GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=1
ZED_CAMERA=1 # ZED SDK 3.0 and above
ZED_CAMERA_v2_8=0 # ZED SDK 2.X

But when I execute this command: 
LD_LIBRARY_PATH =. /: $ LD_LIBRARY_PATH ./uselib data / coco.names cfg / yolov3-tiny.cfg yolov3-tiny.weights zed_camera ,I get the error shown below:

object names loaded 
GStreamer: Error opening bin: no element "zed_camera"
input image or video filename: 
 Video size: [0 x 0]
 exit_flag: detection_data.cap_frame.size = [0 x 0]
 t_cap exit 
 t_prepare exit 
 t_write exit 
 t_network exit 
 t_draw exit 
 t_detect exit 
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /home/nvidia/build_opencv/opencv/modules/highgui/src/window.cpp, line 331
terminate called without an active exception
Aborted (core dumped)

I don't know what to do to solve this problem.Nano comes with CUDN10, cudnn7.5.0, opencv3.3.1.

The second problem:when I test the picture and use the USB camera, the following message appears

”Gtk-Message: 15:29:21.910: Failed to load module "canberra-gtk-module"” 

but the program works normally. No matter which command I use,like
./darknet detector test cfg/voc.data cfg/yolov3-tiny.cfg backup/yolov3-tiny_best.weights
or
LD_LIBRARY_PATH=./:$LD_LIBRARY_PATH ./uselib data/voc.names cfg/yolov3-tiny.cfg backup/yolov3-tiny_best.weights web_camera
the result is the same.
What should I do to solve this two problems,I am looking forward to your reply.thank you very much.

**Comment by Shame-fight** [comment]
> Yes, it should work: https://www.stereolabs.com/blog/announcing-zed-sdk-for-jetson-nano/

Hi,I have compiled and passed darknet on jetson nan. Although there are many warnings, it is successful. The uselib file is generated. My makefile is as follows:

GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=1
ZED_CAMERA=1 # ZED SDK 3.0 and above
ZED_CAMERA_v2_8=0 # ZED SDK 2.X

But when I execute this command: 
LD_LIBRARY_PATH =. /: $ LD_LIBRARY_PATH ./uselib data / coco.names cfg / yolov3-tiny.cfg yolov3-tiny.weights zed_camera ,I get the error shown below:

object names loaded 
GStreamer: Error opening bin: no element "zed_camera"
input image or video filename: 
 Video size: [0 x 0]
 exit_flag: detection_data.cap_frame.size = [0 x 0]
 t_cap exit 
 t_prepare exit 
 t_write exit 
 t_network exit 
 t_draw exit 
 t_detect exit 
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /home/nvidia/build_opencv/opencv/modules/highgui/src/window.cpp, line 331
terminate called without an active exception
Aborted (core dumped)

I don't know what to do to solve this problem.Nano comes with CUDN10, cudnn7.5.0, opencv3.3.1.
I am looking forward to your reply.thank you very much.from your loyal fan.

**Comment by Shame-fight** [comment]
> Yes, it should work: https://www.stereolabs.com/blog/announcing-zed-sdk-for-jetson-nano/

Hi,I have compiled and passed darknet on jetson nan. Although there are many warnings, it is successful. The uselib file is generated. My makefile is as follows:

GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=1
ZED_CAMERA=1 # ZED SDK 3.0 and above
ZED_CAMERA_v2_8=0 # ZED SDK 2.X

But when I execute this command:
LD_LIBRARY_PATH =. /: $ LD_LIBRARY_PATH ./uselib data / coco.names cfg / yolov3-tiny.cfg yolov3-tiny.weights zed_camera ,I get the error shown below:

object names loaded
GStreamer: Error opening bin: no element "zed_camera"
input image or video filename:
Video size: [0 x 0]
exit_flag: detection_data.cap_frame.size = [0 x 0]
t_cap exit
t_prepare exit
t_write exit
t_network exit
t_draw exit
t_detect exit
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /home/nvidia/build_opencv/opencv/modules/highgui/src/window.cpp, line 331
terminate called without an active exception
Aborted (core dumped)

I don't know what to do to solve this problem.Nano comes with CUDN10, cudnn7.5.0, opencv3.3.1.zed_sdk version:ZED_SDK_Tegra_JP42_v3.1.0.run
I am looking forward to your reply.thank you very much.from your loyal fan.

**Comment by Shame-fight** [comment]
> Yes, it should work: https://www.stereolabs.com/blog/announcing-zed-sdk-for-jetson-nano/

can you give me some advice to solve this problem?Looking forward to your reply.
`object names loaded
GStreamer: Error opening bin: no element "zed_camera"
input image or video filename:
Video size: [0 x 0]
exit_flag: detection_data.cap_frame.size = [0 x 0]
t_cap exit
t_prepare exit
t_write exit
t_network exit
t_draw exit
t_detect exit
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /home/nvidia/build_opencv/opencv/modules/highgui/src/window.cpp, line 331
terminate called without an active exception
Aborted (core  dumped)`
@AlexeyAB

**Comment by AlexeyAB** [comment]
Did you compile Darknet with `LIBSO=1  ZED_CAMERA=1` https://github.com/AlexeyAB/darknet/blob/master/Makefile#L8

**Comment by Shame-fight** [comment]
> Did you compile Darknet with `LIBSO=1 ZED_CAMERA=1` https://github.com/AlexeyAB/darknet/blob/master/Makefile#L8

yes,I did it.when I use this command:
`LD_LIBRARY_PATH =. /: $ LD_LIBRARY_PATH ./uselib data / coco.names cfg / yolov3-tiny.cfg yolov3-tiny.weights zed_camera`
whether I connect the zed camera or not,the error will  appear

**Comment by Shame-fight** [comment]
> Did you compile Darknet with `LIBSO=1 ZED_CAMERA=1` https://github.com/AlexeyAB/darknet/blob/master/Makefile#L8

should I carry out this commond:`sudo apt-get install gstreamer1.0`
I I see a bit of a similarity in this issue:[https://stackoverflow.com/questions/31202367/opencv-3-0-0-error-with-gstreamer](url)
@AlexeyAB

**Comment by AlexeyAB** [comment]
> object names loaded GStreamer: Error opening bin: no element "zed_camera" input image or video filename: Video size: [0 x 0] exit_flag:

This error shows that you compiled without `ZED_CAMERA=1` or without `ZED_CAMERA_v2_8=1`
